pkg/middleware/auth_proxy.go

```diff
@@ -30,51 +33,77 @@ func initContextWithAuthProxy(ctx *m.ReqContext, orgID int64) bool {
 		return true
 	}
 
-	query := getSignedInUserQueryForProxyAuth(proxyHeaderValue)
-	query.OrgId = orgID
-	if err := bus.Dispatch(query); err != nil {
-		if err != m.ErrUserNotFound {
-			ctx.Handle(500, "Failed to find user specified in auth proxy header", err)
+	// initialize session
+	if err := ctx.Session.Start(ctx.Context); err != nil {
+		log.Error(3, "Failed to start session", err)
+		return false
+	}
+
+	query := &m.GetSignedInUserQuery{OrgId: orgID}
+
+	// if this session has already been authenticated by authProxy just load the user
+	sessProxyValue := ctx.Session.Get(AUTH_PROXY_SESSION_VAR)
+	if sessProxyValue != nil && sessProxyValue.(string) == proxyHeaderValue && getRequestUserId(ctx) > 0 {
+		query.UserId = getRequestUserId(ctx)
+		if err := bus.Dispatch(query); err != nil {
+			ctx.Handle(500, "Failed to find user", err)
 			return true
 		}
-
-		if !setting.AuthProxyAutoSignUp {
-			return false
+	} else {
+		extUser := &m.ExternalUserInfo{
+			AuthModule: "authproxy",
+			AuthId:     proxyHeaderValue,
 		}
 
-		cmd := getCreateUserCommandForProxyAuth(proxyHeaderValue)
-		if setting.LdapEnabled {
-			cmd.SkipOrgSetup = true
```

Comment:
Is this set somewhere else now? Or is it not needed any more?

Reply:
That is a good question, I don't see it being passed anywhere in the normal ldap flow, but that may be the cause of issues with folks seeing "Cannot remove last organization admin" errors if the ldap user creation flow creates a user org then the org member sync immediately tries to remove them from it (since they'll be the only admin).

Not quite sure what the best solution for that is going to be, for the ldap flow it's fairly straightforward to toggle the flag based on whether there are group membership rules defined, but the way the flow works here it's trickier since we don't run the ldap sync until after the user has already been created.

Reply:
After testing locally, think this might actually be a bug fix for this issue:

https://github.com/grafana/grafana/issues/11311

And I can't find any reasonable test case that fails. The combination of "auth proxy + ldap + auto assign org set to false" seems like it should be very rare.

Reply:
Yeah, I'm not sure which way we want to go with this since it's inconsistent across the 2 methods right now.  I suspect the proper fix for this would be for the authproxy+ldap setup to be changed to use the same flow as the regular ldap rather than creating the user then syncing it.  I started digging into it on Friday and will continue today.

Reply:
I refactored the way authproxy & ldap work together to make things more consistent